### DISCUSSION TITLE
about yolov5 source

### DISCUSSION THREAD

**Discussion by pokaryosy** [ultralytics/yolov5#8239] [Q&A]
Hello,

I would like to ask about the yolov5 source.
I would like to detect objects on my computer screen in realtime. 
Is that possible?
I'm using mac and running at local.

**Comment by glenn-jocher** [comment]
@pokaryosy 👋 Hello! Thanks for asking about **handling inference results**. YOLOv5 🚀 [PyTorch Hub](https://github.com/ultralytics/yolov5/issues/36) models allow for simple model loading and inference in a pure python environment without using `detect.py` including running inference on desktop screenshot contents.

### Simple Inference Example

This example loads a pretrained YOLOv5s model from PyTorch Hub as `model` and passes an image for inference. `'yolov5s'` is the YOLOv5 'small' model. For details on all available models please see the [README](https://github.com/ultralytics/yolov5#pretrained-checkpoints). Custom models can also be loaded, including custom trained PyTorch models and their [exported](https://github.com/ultralytics/yolov5/issues/251) variants, i.e. ONNX, TensorRT, TensorFlow, OpenVINO YOLOv5 models.

```python
import torch

# Model
model = torch.hub.load('ultralytics/yolov5', 'yolov5s')  # yolov5n - yolov5x6 official model
#                                            'custom', 'path/to/best.pt')  # custom model

# Images
im = 'https://ultralytics.com/images/zidane.jpg'  # or file, Path, URL, PIL, OpenCV, numpy, list

# Inference
results = model(im)

# Results
results.print()  # or .show(), .save(), .crop(), .pandas(), etc.

results.xyxy[0]  # im predictions (tensor)
results.pandas().xyxy[0]  # im predictions (pandas)
#      xmin    ymin    xmax   ymax  confidence  class    name
# 0  749.50   43.50  1148.0  704.5    0.874023      0  person
# 2  114.75  195.75  1095.0  708.0    0.624512      0  person
# 3  986.00  304.00  1028.0  420.0    0.286865     27     tie
```
<img width="550" src="https://user-images.githubusercontent.com/26833433/149679662-b2021c78-59b2-4f52-b768-1e5161537c85.png">

See [YOLOv5 PyTorch Hub Tutorial](https://github.com/ultralytics/yolov5/issues/36) for details.

Good luck 🍀 and let us know if you have any other questions!

**Reply by pokaryosy** [reply]
@glenn-jocher 
Thank you for your reply.

I mean,
Could I select the source like the below?
$ python path/to/detect.py --weights yolov5s.pt --source 0 ←select computer screen in realtime
Is that possible? or are there another method?

**Reply by glenn-jocher** [reply]
@pokaryosy see [YOLOv5 PyTorch Hub Tutorial](https://github.com/ultralytics/yolov5/issues/36) for details on screenshot inference.

**Reply by pokaryosy** [reply]
@glenn-jocher 
Thank you for your reply.
Just to confirm, I don't want to detect only a single screenshot, I want to detect what is continuously on the screen, like object detection with a webcam.
Would that be solved by the tutorial, which is in the link?

**Reply by glenn-jocher** [reply]
yes

**Comment by pokaryosy** [comment]
It seems that the information he shared is not enough to solve the problem, but can anyone else come up with a solution?

**Reply by cgotschlich** [reply]
If you want to run inference on your own computer's screen in real time you would have to stream your own screen via RTSP, RTMP or HTTP, which is not built into YOLOv5, and then pass the address to you stream e.g. `http://127.0.0.1:8090/screen` to the `detect()` function, as you figured out earlier: 
```
python detect.py --source 'http://127.0.0.1:8090/screen'
```

**Reply by glenn-jocher** [reply]
@pokaryosy @currychris see [YOLOv5 PyTorch Hub Tutorial](https://github.com/ultralytics/yolov5/issues/36) for details on screenshot inference.

<img width="891" alt="Screen Shot 2022-07-29 at 3 14 55 PM" src="https://user-images.githubusercontent.com/26833433/181768346-ca1b6a6c-652e-4c9e-9589-9f5c174462d7.png">

**Comment by JMChenl** [comment]
Hello author, please ask which version can use YOLOV5 6.1 or 6.2 to do detection to achieve the purpose of tracking

**Reply by glenn-jocher** [reply]
@JMChenl 👋 Hello! Thanks for asking about object **tracking** in computer vision. All releases of YOLOv5 are designed for detection, with 6.2 now also capable of classification.

YOLOv5 🚀 is an object **detector** that detect, localizes and classifies objects in a single image. It does not connect objects across multiple images, for this you need a tracking solution. A few possible tracking solutions are:

- [Extended Kalman Filter (EKF)](https://en.wikipedia.org/wiki/Kalman_filter)
- [Kanade-Lukas-Tomasi (KLT) Feature Tracker](https://en.wikipedia.org/wiki/Kanade%E2%80%93Lucas%E2%80%93Tomasi_feature_tracker)
- [YOLOv5 StrongSort Tracker](https://github.com/mikel-brostrom/Yolov5_StrongSORT_OSNet) 
- [Apple iOS VNTrackObjectRequest](https://developer.apple.com/documentation/vision/tracking_multiple_objects_or_rectangles_in_video)

Good luck 🍀 and let us know if you have any other questions!

**Reply by pokaryosy** [reply]
Hello @JMChenl !
I'm using Yolov5 not 6.

**Comment by JMChenl** [comment]
Hello @YOSY POKARI！
YOLOV5‘s v6.0

发自我的iPhone
